Client/Frontend/Browser/BrowserViewController/BrowserViewController+WKNavigationDelegate.swift

```diff
@@ -230,11 +220,16 @@ extension BrowserViewController: WKNavigationDelegate {
        navigationAction.targetFrame?.isMainFrame == true,
        let redirectURL = WebsiteRedirects.redirect(for: url) {
       
-      decisionHandler(.cancel, preferences)
       tab?.loadRequest(URLRequest(url: redirectURL))
-      return
+      return (.cancel, preferences)
```

Comment:
This technically chnages the order of execution, can we make sure that this change hasn't broken anything?

Reply:
I tried it for debouncing and it's fine. However if we go for async await we would have to do this change regardless.

Reply:
So I tested this and it works just fine

Reply:
Do you think it would make sense to wrap the `loadRequest` in a `Task { @MainThread in }`

Reply:
Hopping it to next runtime loop would be closer to what we have now but probably unnecessary if it was tested and working